Lib/test/test_exceptions.py

```diff
@@ -953,6 +953,71 @@ def __del__(self):
             pass
         self.assertEqual(e, (None, None, None))
 
+    def test_no_hang_on_context_chain_cycle1(self):
+        # See issue 25782.
+
+        def cycle():
+            try:
+                raise ValueError(1)
+            except ValueError as ex:
+                ex.__context__ = ex
+                raise TypeError(2)
+
+        try:
+            cycle()
+        except Exception as e:
+            exc = e
+
+        self.assertIsInstance(exc, TypeError)
+        self.assertIsInstance(exc.__context__, ValueError)
```

Comment:
Does it make sense also to check `exc.__context__.__context__` (to check that the cycle was preserved)?